src/http-adapter/index.js

```diff
@@ -0,0 +1,203 @@
+   * @returns {string} - the built url string relating to the JSON:API entity
+   */
+  buildUrl (type, id) {
+    let idPath = `/${id}`
+    let typePath = `/${type}`
+    let path = this.urlPrefix || ''
+
+    path += typePath
+
+    if (id != null) {
+      path += idPath
+    }
+
+    return `${path}/`
+  }
+
+  /**
+   * Get entity by link.
+   * @param {string} link - Link to entity.
+   * @param {Object} query - query patams for a given request in a key/value pair format
+   * @returns {Promise} Promise for fetched data.
+   */
+  async get (link, query) {
+    return this._http(HTTP_METHOD.GET, link, query)
+  }
+
+  /**
+   * Create entity.
+   * @param {string} link - Entity url.
+   * @param {Object} payload - Data to create entity with.
+   * @returns {Promise} Promise for created data.
+   */
+  async create (link, payload) {
+    return this._http(HTTP_METHOD.POST, link, payload)
+  }
+
+  /**
+   * Update entity.
+   * @param {string} link - Entity url.
+   * @param {Object} payload - Data to update entity with.
+   * @returns {Promise} Promise for updated data.
+   */
+  async update (link, payload) {
+    return this._http(HTTP_METHOD.PATCH, link, payload)
+  }
+
+  /**
+   * Destroy entity.
+   * @param {string} link - Entity self link.
+   * @returns {Promise} Promise for destroy.
+   */
+  async destroy (link) {
+    return this._http(HTTP_METHOD.DELETE, link)
+  }
+
+  /**
+   * Make a generic request that requires more options then CRUD methods provided.
+   * Does not set default Accept or Content-Type. This must be provided!
+   * @param {string} url - URL where the request is being sent.
+   * @param {Object} options - any options the user wants to pass to the request.
+   * this can contain method, headers, internal, and data
+   * @returns {Promise} Promise relating to request.
+   */
+  async request (
+    url,
+    options = {}
+  ) {
+    options.method = options.method || HTTP_METHOD.GET
+    options.headers = options.headers || {}
+    // set internal to true if this request in intended to go to an interal server to gain the default headers passed into the constructor
+    options.internal = options.internal || false
+
+    // pass in data, even if none is defined to preverve _http method
+    // pass in headers, even if none are defined to override default headers that are set with the _http method
+    return this._http(
+      options.method,
+      url,
+      options.data,
+      options.headers,
+      options.internal
+    )
+  }
+
+  /**
+   * Check if requestion serialization is turned on. If so, the current request will
+   * wait for previous requests to settle. Otherwise, the request will just be executed
+   * @private
+   * @param {Object} -  options passed via destructoring, which can contain url,
+   * method, headers, data, and isInternal properties
+   * @returns {Promise} Promise relating to request resolution.
+   */
+  async _checkSerializeRequests ({
```

Comment:
Does this function as a protected method for the extending class? might we worth a try to check and make sure

Reply:
I might save this one for a later date. Once we get babel upgraded in this repository to prevent testing failures with this feature, we can then plan to incorporate it elsewhere